README.md

```diff
@@ -508,9 +508,43 @@ In addition to the container runtime daemons, these credential settings will als
 * `settings.network.hostname`: The desired hostname of the system.
   **Important note for all Kubernetes variants:** Changing this setting at runtime (not via user data) can cause issues with kubelet registration, as hostname is closely tied to the identity of the system for both registration and certificates/authorization purposes.
 
-Most users don't need to change this setting as the following defaults work for the majority of use cases.
-If this setting isn't set we attempt to use DNS reverse lookup for the hostname.
-If the lookup is unsuccessful, the IP of the node is used.
+  Most users don't need to change this setting as the following defaults work for the majority of use cases.
+  If this setting isn't set we attempt to use DNS reverse lookup for the hostname.
+  If the lookup is unsuccessful, the IP of the node is used.
+
+* `settings.network.hosts`: A mapping of IP addresses to domain names which should resolve to those IP addresses.
+   This setting results in modifications to the `/etc/hosts` file  for Bottlerocket.
+   Note that this setting does not typically impact name resolution for containers, which usually rely on [orchestrator-specific](https://docs.aws.amazon.com/AmazonECS/latest/APIReference/API_HostEntry.html) [mechanisms](https://kubernetes.io/docs/tasks/network/customize-hosts-file-for-pods/) for configuring static resolution.
+   
+   Example:
+   ```toml
+   [settings.network]
+   hosts = [
+    ["10.0.0.0", ["test.example.com", "test1.example.com"]],
+    ["10.1.1.1", ["test2.example.com"]]
+   ]
+   ```
+   This example would result in an `/etc/hosts` file entries like so:
+   ```
+   10.0.0.0 test.example.com test1.example.com
+   10.1.1.1 test2.example.com
+   ```
+   Repeated entries are merged (including loopback entries), with the first aliases listed taking precedence. e.g.:
+
+   ```toml
+   [settings.network]
+   hosts = [
+    ["10.0.0.0", ["test.example.com", "test1.example.com"]],
+    ["10.1.1.1", ["test2.example.com"]],
+    ["10.1.1.0", ["test3.example.com"]],
+   ]
+   ```
+   Would result in `/etc/hosts` entries like so:
+   ```
+   10.0.0.0 test.example.com test1.example.com test3.example.com
+   10.1.1.1 test2.example.com
+   ```
```

Comment:
Hi I'm trying to validate if the current logic will work for our use-case.

1. Why in the example test3.example.com (`10.1.1.0`) is merged with the `10.0.0.0` line and thus effectively rewritten to `10.0.0.0`?

2. Is it possible with the current logic to have one hostname to resolve to multiple IPs? The input syntax seems to support it. Just want to make sure no de-duplication will happen in this case. We will require a final `/etc/hosts` file that looks something like this:

```
10.0.2.252 test1.example.com
10.0.2.236 test1.example.com
10.0.3.205 test1.example.com
```

Reply:
Hi, thanks for taking a look at this PR.

1. This is a documentation mistake, I meant for `10.1.1.0` to be a repeat of `10.0.0.0`. I'll fix that.
2. Yes, you can have one hostname resolve to multiple IPs. De-duplication will only occur for IPs, not hostnames. If you submit something like the following, you should end up with `/etc/hosts` entries like the ones you've given:

```
hosts = [
    ["10.0.2.252", ["test1.example.com"]],
    ["10.0.2.236", ["test1.example.com"]],
    ["10.0.3.205", ["test1.example.com]]
]
```